java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -76,4 +86,39 @@ private static boolean hasPublicModifier(MethodTree methodTree) {
     return ModifiersUtils.hasModifier(methodTree.modifiers(), Modifier.PUBLIC);
   }
 
+  private static boolean hasCompliantGeneratedConstructors(ClassTree classTree) {
+    if (classTree.modifiers().annotations().stream().noneMatch(UtilityClassWithPublicConstructorCheck::isLombokConstructorGenerator)) {
+      return false;
+    }
+    return classTree.modifiers().annotations().stream()
+      .filter(UtilityClassWithPublicConstructorCheck::isLombokConstructorGenerator)
+      .noneMatch(UtilityClassWithPublicConstructorCheck::hasPublicAccess);
+  }
+
+  private static boolean isLombokConstructorGenerator(AnnotationTree annotation) {
+    return LOMBOK_CONSTRUCTOR_GENERATORS.contains(annotation.annotationType().symbolType().fullyQualifiedName());
+  }
+
+  private static boolean hasPublicAccess(AnnotationTree annotation) {
+    for (var arg : annotation.arguments()) {
+      var assignment = (AssignmentExpressionTree) arg;
+      if (isAccessLevelNotPublic(assignment.expression())) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
Can be simplified to:

```suggestion
    return annotation.arguments().stream().noneMatch(it ->
        isAccessLevelNotPublic(((AssignmentExpressionTree) it).expression())
    );
```